### PR TITLE
Disambiguate device model

### PR DESCRIPTION
**Pull request by tdsmith**
"Device model" can mean a couple different things on Android (e.g. `Build.DEVICE` vs `Build.MODEL`); this clarifies that experiments match on the former.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

cc @travis79 for review!

### CODE REVIEW

File: components/service/experiments/README.md [hunk 1]
```diff
@@ -177,32 +177,36 @@ An individual experiment record looks e.g. like this:
...
+| match                     | object |     x    | Object containing the filter parameters to match specific user groups.                                                                          |                                                                 |
+| match.app_id              | regex  |          | The app ID (package name)                                                                                                                       | `"^org.mozilla.firefox_beta${'$'}|^org.mozilla.firefox${'$'}"`  |
+| match.device_manufacturer | regex  |          | The Android device manufacturer.                                                                                                                |                                                                 |
+| match.device_model        | regex  |          | The Android device model, [`Build.DEVICE`]. This is the codename for the device's industrial design.                                            | `"^(flame|coral)$"` (to match Pixel 4 or Pixel 4 XL)            |
```
Comment (travis79) [inline components/service/experiments/README.md]: The pipe in between `flame|coral` is breaking the table.  Nothing shows up after this pipe.

File: components/service/experiments/README.md [hunk 2]
```diff
@@ -177,32 +177,36 @@ An individual experiment record looks e.g. like this:
...
+| match.locale_country      |        |          | The default locales country.                                                                                                                    | `"USA|DEU"`                                                     |
```
Comment (travis79) [inline components/service/experiments/README.md]: Same as above, the pipe in between `USA|DEU` is breaking the markdown table.

File: components/service/experiments/README.md [hunk 3]
```diff
@@ -177,32 +177,36 @@ An individual experiment record looks e.g. like this:
...
+| match.locale_language     |        |          | The default locales language.                                                                                                                   | `"eng|zho|deu"`                                                 |
```
Comment (travis79) [inline components/service/experiments/README.md]: Same as above, extra pipes in the example are breaking the markdown table.

**Comment by tdsmith** [comment]
Oops; thanks for looking. I had hoped the backticks would save me.

**Pull request merged**